app/src/main/java/com/example/trainingintive/dog_images_feature/data/repository/DogImageRepository.kt

```diff
@@ -24,4 +30,16 @@ class DogImageRepository @Inject constructor(
     fun update(imageUrl: DogImageUrl) = dogImageDao.update(imageUrl.toDatabaseEntity())
 
     fun remove(imageUrl: DogImageUrl) = dogImageDao.remove(imageUrl.toDatabaseEntity())
+
+    fun update(imageUrls: List<DogImageUrl>) = dogImageDao.update(imageUrls.map { it.toDatabaseEntity() })
+
+    private fun getLastPosition() = dogImageDao.count()
+
+    fun changeImagePosition(urls: List<DogImageUrl>, from: Int, to: Int): Completable {
```

Comment:
If you are doing such operations (it will took some time) it will be best to wrap whole function with Completable to execute this code on other thread